Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,598 @@
+
+```swift
+struct User: Identifiable {
+  var id: String
+  var name: String
+  var lastname: String
+  var email: String
+}
+
+struct StatusInformation {
+  var uptime: UInt64
+  // ...
+}
+
+struct TestService: WebService {
+  @PathParameter var userId: User.ID
+
+  var content: some Component {
+    Group("status") {
+      StatusInformationHandler()
+    }
+    Group("user", $userId) {
+      UserHandler(userId: $userId)
+    }
+  }
+}
+```
+
+It will be observed that the `StatusInformationHandler` returns instances of type `StatusInformation`.
+So when another Endpoint references the type `StatusInformation`, we can easily get a reference to the Endpoint returning
+that type.
+
+Furthermore, if we detect, that the return type of a `Handler` conforms to the `Identifiable` protocol, there are some
+additional steps we need to check. 
+* If we can match the type of the `.id` property to any `@PathParameter` (with the same type) contained in the path,
+    we save that return type to our index (Though: see note below).
+* We we can't match the `.id` property to any `@PathParameter` the `Handler` will be ignored and not indexed.
+Instead such endpoints can be used as a source of relationship information
+(see [3.3.](#331-implicitly)).
+
+Some special cases:
+* Endpoints containing a `@PathParameter` in their path, but not returning a type
+ conforming `Identifiable` in their `handle()` (meaning we can't match a property to the path parameter), will not be indexed.
+* In a case where multiple `Handler` return the same type (e.g. multiple handlers returning `StatusInformation`)
+non of them are added to the index. Instead we rely on the user to mark the appropriate `Handler`,
+see [3.2.](#32-using-defaultreference-to-resolve-ambiguous-return-type-information).
+
+_Note: As the type of a `@PathParameter` is a `LosslessStringConvertible` it most certainly is either a 
+`String` or an `Int`. Meaning, checking if the type of the `User.id` property is equal to `User.ID` type might
+not be strong enough.
+Thus it could be necessary for `@PathParemeter`s to specify the type: e.g. `@PathParameter(type: User.self)` where
+the `type` argument must conform to the `Identifiable` protocol (or something similar, as `Identifiable.ID` conforms 
+to `Hashable` and we actually need `LosslessStringConvertible`)_
+
+### 3.2. Using `.defaultReference` to resolve ambiguous return type information
+
+As describe in [3.1.](#31-indexing-endpoints-by-their-handler-return-type) Apodini won't index type information
+if it isn't unambiguous.
+
+In such cases where there are multiple `Handler` returning the same type, the user can use `.defaultReference` to
+explicitly mark which Endpoint should be indexed for the given return type.  
+
+```swift
+struct StatusInformation {
+  var uptime: UInt64
+  // ...
+}
+// ...
+var content: some Component {
+  Group("status") {
+    StatusInformationHandler()
+      .defaultReference()
```

Comment:
I would suggest to use a common naming across all modifiers. E.g. either relationship, reference or link. I am currently trending towards relationship but all other options would also be fine with me.

Reply:
Resolved. See https://github.com/Apodini/Apodini/pull/41#issuecomment-748679123.